envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/SourceIpBasedAuthenticationTest.kt

```diff
@@ -0,0 +1,73 @@
+package pl.allegro.tech.servicemesh.envoycontrol
+
+import okhttp3.Headers
+import okhttp3.Response
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import pl.allegro.tech.servicemesh.envoycontrol.config.Envoy1Ads
+import pl.allegro.tech.servicemesh.envoycontrol.config.Envoy2Ads
+import pl.allegro.tech.servicemesh.envoycontrol.config.EnvoyControlRunnerTestApp
+import pl.allegro.tech.servicemesh.envoycontrol.config.EnvoyControlTestConfiguration
+import pl.allegro.tech.servicemesh.envoycontrol.config.envoy.EnvoyContainer
+
+internal class SourceIpBasedAuthenticationTest : EnvoyControlTestConfiguration() {
+
+    companion object {
+
+        private val properties = mapOf(
+            "envoy-control.envoy.snapshot.incoming-permissions.enabled" to true,
+            "envoy-control.envoy.snapshot.incoming-permissions.source-ip-authentication.enabled-for-services" to
+                    listOf("echo"),
+            "envoy-control.envoy.snapshot.routes.status.create-virtual-cluster" to true,
+            "envoy-control.envoy.snapshot.routes.status.path-prefix" to "/status/",
+            "envoy-control.envoy.snapshot.routes.status.enabled" to true
+        )
+
+        @JvmStatic
+        @BeforeAll
+        fun setupTest() {
+            setup(appFactoryForEc1 = { consulPort ->
+                EnvoyControlRunnerTestApp(properties = properties, consulPort = consulPort)
+            }, envoyConfig = Envoy1Ads, secondEnvoyConfig = Envoy2Ads, envoys = 2)
+        }
+    }
+
+    @Test
+    fun `should allow access to selected clients using source based authentication`() {
```

Comment:
Can we add test which check if access has been denied?

Reply:
So you want to split this test to two right? (note `validReponse` and `invalidResponse`).

Reply:
@MarcinFalkowski @pzmi do you think we should go with 2 tests or 1? (my motivation behind two assertions in integration tests is that they are costlier and either way do not point, like unit tests, exactly where the issue is, but if everyone agrees that we want 2 tests then I'm ok with that)

Reply:
I think it should be one test. If we split it to 2 tests, there is a better chance we would get false positives. For example if we would have a test which would only test if access is denied, it will pass when something is misconfigured and Envoy returns bad response always. When we check valid and invalid response in one test, we are sure, that invalid response was caused by access denied.

Reply:
Is that ok Łukasz?